src/fluka/fluka_funcs.cpp

```diff
@@ -0,0 +1,1104 @@
+    state.on_boundary = false;
+    retStep = mulbou_.tsense;
+    // fluka takes care of reseting newreg outisde of ourtine
+    //    proposed_step = mulbou_.tsense;
+    if(debug)
+      std::cout << "This is a sensing step " << mulbou_.tsense << std::endl;
+  }
+  */
+
+
+  if (debug)  {
+    std::cout << "Region on other side of surface is  = " << newRegion
+              << ", Distance to next surf is " << retStep << std::endl;
+  }
+
+  // if lsense true, a false sensing step, facet not hit, surface not crossed
+  // if sensing step takes is beyond volume doesnt matter, if takes in same volume
+  // doesnt matter
+  //  if(mulbou_.lsense) {
+  //    state.history.rollback_last_intersection();
+  //    if(debug) {
+  //      std::cout << "This is a sensing step " << mulbou_.tsense << std::endl;
+  //    }
+  //    newRegion = -3;
+  //  }
+
+  // save all the state we need
+  state.old_direction[0] = dir[0];
+  state.old_direction[1] = dir[1];
+  state.old_direction[2] = dir[2];
+
+  // position
+  state.old_position[0] = point[0];
+  state.old_position[1] = point[1];
+  state.old_position[2] = point[2];
+
+
+  return;
+}
+
+// function to determine the angle between
+double dot_product(moab::EntityHandle surface, double point[3], double direction[3]) {
+  moab::EntityHandle rval;
+  double normal[3] = {0, 0, 0};
+  if (debug) {
+    std::cout << "dot_product on surface " << surface << std::endl;
+  }
+  rval = DAG->get_angle(surface, point, normal);
+  if (moab::MB_SUCCESS != rval)
+    fludag_abort("dot_product", "DAGMC failed in get_angle", rval);
+
+  //may want to write our own inline vec angle function
+  moab::CartVect v1(normal);
+  moab::CartVect v2(direction);
+
+  double angle = moab::angle(v1, v2);
+
+  if (debug) {
+    std::cout << "------ dot_product -------" << std::endl;
+    std::cout << "angle between surface, " << surface << "with normal, " << v1 << std::endl;
+    std::cout << " amd direction " << v2 << " is " << angle << std::endl;
+  }
+
+  return angle;
+}
+
+/* print state */
+void print_state(particle_state& state) {
+  std::cout << "-------print_state-------" << std::endl;
+  std::cout << "pos= " << state.old_position[0] << " " << state.old_position[1];
+  std::cout << " "     << state.old_position[2] << std::endl;
+  std::cout << "dir= " << state.old_direction[0] << " " << state.old_direction[1];
+  std::cout << " "     << state.old_direction[2] << std::endl;
+
+  std::cout << "next_surface: " << state.next_surface << " prev_surface: " << state.prev_surface << std::endl;
+  std::cout << "prevRegion: " << state.PrevRegion  << std::endl;
+  std::cout << "on_boundary: " << state.on_boundary << std::endl;
+  return;
+}
+
+/* resets state */
+void reset_state(particle_state& state) {
+  if (debug) {
+    std::cout << "-----reset_state----" << std::endl;
+    std::cout << "Resetting state....." << std::endl;
+  }
+
+  // set position to some large value
+  state.old_position[0] = 9.e99;
```

Comment:
If you are looking to max out this value (assuming it is a double), you can use `std::numeric_limits<double>::max()`.

[1] https://stackoverflow.com/a/5834704/1015951